*** Socket.h.orig	2015-04-27 13:47:49.830762700 -0700
--- Socket.h	2015-04-27 13:57:09.911546800 -0700
***************
*** 2,9 ****
  * Socket.h
  */
  
! #ifdef _WIN32
! 	#include <afxsock.h>		/* MFC socket extensions */
  #endif
  
  int  ConnectToServer (char *Ip_Address, int Ip_Port, double TimeOut);
--- 2,9 ----
  * Socket.h
  */
  
! #ifdef __cplusplus
! extern "C" {
  #endif
  
  int  ConnectToServer (char *Ip_Address, int Ip_Port, double TimeOut);
***************
*** 12,14 ****
--- 12,26 ----
  void CloseSocket (int SocketID);
  char * GetError (int SocketID);
  void strncpyWithEOS(char * szStringOut, const char * szStringIn, int nNumberOfCharToCopy, int nStringOutSize);
+ 
+ /* MX functions for enabling or disabling communication socket debugging. */
+ 
+ int  mxp_newport_xps_get_comm_debug_flag(void);
+ void mxp_newport_xps_set_comm_debug_flag(int debug_flag);
+ double mxp_newport_xps_get_comm_delay(void);
+ void mxp_newport_xps_set_comm_delay(double delay);
+ 
+ #ifdef __cplusplus
+ }
+ #endif
+ 
