*** Socket.h.orig	2015-04-27 13:47:49.830762700 -0700
--- Socket.h	2015-04-27 13:57:09.911546800 -0700
***************
*** 2,9 ****
  * Socket.h
  */
  
! #ifdef _WIN32
! 	#include <afxsock.h>		/* MFC socket extensions */
  #endif
  
  int  ConnectToServer (char *Ip_Address, int Ip_Port, double TimeOut);
--- 2,9 ----
  * Socket.h
  */
  
! #ifdef __cplusplus
! extern "C" {
  #endif
  
  int  ConnectToServer (char *Ip_Address, int Ip_Port, double TimeOut);
***************
*** 12,14 ****
--- 12,21 ----
  void CloseSocket (int SocketID);
  char * GetError (int SocketID);
  void strncpyWithEOS(char * szStringOut, const char * szStringIn, int nNumberOfCharToCopy, int nStringOutSize);
+ 
+ #include "Socket_MX.h"
+ 
+ #ifdef __cplusplus
+ }
+ #endif
+ 
