*** Socket.cpp.orig	2014-05-14 19:23:57.873814416 -0500
--- Socket.cpp	2014-05-14 19:38:49.904597286 -0500
***************
*** 1,6 ****
--- 1,11 ----
  /*///////////////////////////////////////////////////////////////////////////////
  * Socket.cpp
  */
+ 
+ #if !defined(_WIN32)
+ #  include "Socket_MX.h"
+ #endif
+ 
  #include "Socket.h" 
  
  #define TIMEOUT 300
***************
*** 73,79 ****
  		TimeoutSocket[socketID]	= TIMEOUT;
  
  	/* Socket array */
! 	struct fd_set *Sockets = new struct fd_set;
  	FD_ZERO(Sockets);
  	FD_SET(m_sConnectSocket[socketID], Sockets);
  
--- 78,84 ----
  		TimeoutSocket[socketID]	= TIMEOUT;
  
  	/* Socket array */
! 	fd_set *Sockets = new fd_set;
  	FD_ZERO(Sockets);
  	FD_SET(m_sConnectSocket[socketID], Sockets);
  
